core/clients/token_flow.go

```diff
@@ -73,5 +60,5 @@ func (c *TokenFlow) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, fmt.Errorf("please run Init()")
 	}
 	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.config.ServiceAccountToken))
-	return Do(c.client, req, c.config.ClientRetry)
+	return c.client.Do(req)
```

Comment:
I'm not sure if it would be more correct to call `c.client.Transport.RoundTrip(req)` here.

I think it's more common to call `RoundTrip` here, and that should basically be `Do` + some extra stuff. But that runs any custom code introduced by the user in the `RoundTripper` interface, which may lead to unintended changes.

Since the deleted `Do` routine was calling `client.Do`, I think it's best to keep that here. Otherwise we may introduce unintended changes. But should be kept in mind

Reply:
as discussed offline, `Do` calls `Roundtrip`, so we'll leave it like this